python/events/evaluation_tools/events/event_detection/decomposition.py

```diff
@@ -87,10 +89,44 @@ def detrend_streamflow(
     # Return detrended series
     return detrended
 
+def event_boundaries(event_points: pd.Series):
+    """Return a two column pandas.DataFrame with 'start' and 'end' event times
+        generated from a time series of boolean event points.
+        
+        Parameters
+        ----------
+        event_points: pandas.Series
+            Boolean time series where True indicates the associated value
+            in the `series` is part of an event.
+
+        Returns
+        -------
+        events: pandas.DataFrame
+            A two column DataFrame with a row for each event detected. `start` and 
+            `end` columns indicate the boundaries of each event.
+        
+        """
+    # Identify event starts
+    forward_shift = event_points.shift(1).fillna(False)
+    starts = (event_points & ~forward_shift)
+    starts = starts[starts]
+
+    # Identify event ends
+    backward_shift = event_points.shift(-1).fillna(False)
+    ends = (event_points & ~backward_shift)
+    ends = ends[ends]
+
+    # Extract times
+    return pd.DataFrame({
```

Comment:
It seems like it is possible for `starts` and `ends` to be of different length. I am probably wrong about this, but is that true? If, so this should throw an error.

Reply:
I don't think this is possible. Both `starts` and `ends` are generated from the same `Series`. `shift` changes the index of the values, but does not alter the index. This is why both calls to `shift` are followed by `fillna`. The index is unchanged and the front and back of the new series have `NaN`.

Reply:
Okay, that makes sense. Disregard my comment.